### ISSUE TITLE
Inquiry About Releasing the Training Code

### ISSUE THREAD

**Issue by Jiahao-Ma** [yyfz/Pi3#16]
Impressive work! Are you planning to release the training code?

**Comment by yyfz** [comment]
Hi, [Jiahao-Ma](https://github.com/Jiahao-Ma)!

We will release the training code after our paper is accepted.

**Comment by HaisongDing** [comment]
Can you share more details on the hyper-parameters, for example threshold for BCE loss, loss weight for each loss items?

**Comment by yyfz** [comment]
We set the BCE loss weight for the confidence score to 0.05. It's important to note that the confidence head is trained in a separate stage. For this stage, we freeze the model's backbone and train only the confidence-related parts. After the initial training of the main model is complete, this second stage for the confidence score converges very rapidly, achieving good performance within just a few epochs.

To summarize the overall loss weights: the points loss is weighted by 1, the translation loss by 10, and the rotation loss by 0.1. The confidence score is handled in a separate training process where its loss is weighted by 0.05.

I apologize for omitting these details in the paper. We will add a full explanation in a future updated version on arXiv.

**Comment by HaisongDing** [comment]
Thanks, how about the normal loss weight?

**Comment by yyfz** [comment]
> Thanks, how about the normal loss weight?

The normal loss is adapted from MoGe, its weight is set to 1.0.

**Comment by selflein** [comment]
Thank you for the details!

How is the ground truth scene scaled? 

If I get that correct that will affect the loss scale as the predicted pointmap will be scaled to the ground truth scale which will in turn affect the values of the losses.

Also do you clip the scale to be within a reasonable range to not let the predicted pointmap be very small or very large in magnitude? 

Thank you!

**Comment by matthew-strong-ai** [comment]
Thanks @yyfz . I have a quick question about the point map loss -- do you perform loss on the local or global point maps?

**Comment by yyfz** [comment]
> Thanks [@yyfz](https://github.com/yyfz) . I have a quick question about the point map loss -- do you perform loss on the local or global point maps?

We perform loss on local pointmaps, we don't predict global pointmaps.

**Comment by matthew-strong-ai** [comment]
In the meantime, I will try to replicate the training code, and potentially open source it when done.

**Comment by matthew-strong-ai** [comment]
oh that is great! Excited to hear back

**Comment by matthew-strong-ai** [comment]
LFG! Thanks @yyfz
